### DISCUSSION TITLE
Bot as systemd daemon breaks

### DISCUSSION THREAD

**Discussion by Floweynt** [jagrosh/MusicBot#894] [Troubleshooting] [answered]
Trying to run this as a systemd daemon causes it to break
music.service:
```ini
[Unit]
Description=music bot
After=network.target
StartLimitIntervalSec=0
[Service]
Type=simple
Restart=always
RestartSec=1
User=server
ExecStart=/usr/bin/env java -Dnogui=true -Dconfig=/home/server/music-bot/config.txtjar /home/server/music-bot/JMusicBot-0.3.5.jar

[Install]
WantedBy=multi-user.target
````
This causes:
`Sep 21 22:13:31 env[553866]: Error: Could not find or load main class .home.server.music-bot.JMusicBot-0.3.5.jar`

**Comment by Sanduhr32** [comment, marked as answer]
`ExecStart=/usr/bin/env java -Dnogui=true -Dconfig=/home/server/music-bot/config.txtjar /home/server/music-bot/JMusicBot-0.3.5.jar`

Should be:

`ExecStart=/usr/bin/env java -Dnogui=true -Dconfig=/home/server/music-bot/config.txt -jar /home/server/music-bot/JMusicBot-0.3.5.jar`
You also probably want to set stuff like

```ini
[Unit]
After=network.target network-online.target
Requires=network.target network-online.target
[Service]
WorkingDirectory=/home/server/music-bot
```

Hope I could help.

**Comment by JulianGmp** [comment]
This might seem a bit off topic but I can also suggest the following additional settings for the service file:
```service
[Unit]
StartLimitIntervalSec=500
StartLimitBurst=5

[Service]
Restart=on-failure
RestartSec=5s
```

As you can tell this will automatically restart the service after 5 seconds if it happens to exit with a crash.

**Reply by Sanduhr32** [reply]
This is planned / implemented on a currently open PR documenting how to use systemd for autostarting the bot on linux.
